frame/benchmarking/src/lib.rs

```diff
@@ -695,6 +799,110 @@ macro_rules! benchmark_backend {
 	};
 }
 
+// Creates #[test] functions for the given bench cases.
+#[macro_export]
+#[doc(hidden)]
+macro_rules! impl_bench_case_tests {
+	(
+		{ $module:ident, $new_test_exec:expr, $exec_name:ident, $test:path, $extra:expr }
+		{ $( $names_extra:tt )* }
+		$( { $( $bench_inst:ident )? } $bench:ident )*
+	)
+	=> {
+		$crate::impl_bench_name_tests!(
+			$module, $new_test_exec, $exec_name, $test, $extra,
+			{ $( $names_extra )* },
+			$( { $bench } )+
+		);
+	}
+}
+
+// Creates a #[test] function for the given bench name.
+#[macro_export]
+#[doc(hidden)]
+macro_rules! impl_bench_name_tests {
+	// recursion anchor
+	(
+		$module:ident, $new_test_exec:expr, $exec_name:ident, $test:path, $extra:expr,
+		{ $( $names_extra:tt )* },
+		{ $name:ident }
+	) => {
+		$crate::paste::paste! {
+			#[test]
+			fn [<bench_ $name>] () {
+				$new_test_exec.$exec_name(|| {
+					// Skip all #[extra] benchmarks if $extra is false.
+					if !($extra) {
+						let disabled = $crate::vec![ $( stringify!($names_extra).as_ref() ),* ];
+						if disabled.contains(&stringify!($name)) {
+							$crate::log::error!(
+								"INFO: extra benchmark skipped - {}",
+								stringify!($name),
+							);
```

Comment:
it is a bit weird to log an error for an info
```suggestion
							$crate::log::info!(
								"extra benchmark skipped - {}",
								stringify!($name),
							);
```

Reply:
I think this is so the log appears when running the tests by default.

Reply:
I feel like if somebody writes `extra = false` then he expects the extra benchmarks to not run. So logging something doesn't look very useful.
Actually I don't know why somebody would write `extra = false` to be honest.

Reply:
Yeah I guess only if one of the extra benchmarks is so insane heavy... but yeah, doesnt really make sense to have `extra = false` ever in tests. That is the point of tests.

Reply:
I see, then ok

Reply:
I wondered the same, but thought that you probably only output the error stream.  
Resolving for now.